src/client/redux/reducers/products.js

```diff
@@ -1,6 +1,8 @@
+import * as actionTypes from "../actionTypes";
+
 export default function productsReducer(state = [], action) {
   switch (action.type) {
-    case "SET_PRODUCTS":
+    case actionTypes.GET_PRODUCTS_SUCCESS:
```

Comment:
Should we store products data as object instead of array for easy access
productsState {
     loading: false,
     products: { 
        1: {id: 1, name: 'laptop'},
        2: {id: 2, name: 'keyboard'}
     },  // for easy access of products in detailed view
     productIds: [1, 2]  // to retain order of products fetched
}

Reply:
The suggested state modelling is totally correct, i would like to just add few more flags to it

```initialState =  {
   byId:{
   },
   id: [],
   isLoading: false,
   isError: false,
}```

Reply:
@sumit-gupta91 Add the complete code which you demonstrate.  I will merge it into develop branch and later rename it to solutions.
@sumit-gupta91 
During the workshop attendees will work from master branch which has only the boilerplate code.
At the end, we can point them to solutions branch